lib/schemas.js

```diff
@@ -129,9 +129,9 @@ function generateFluentSchema (schema) {
   }
 }
 
-function getSchemaAnyway (schema, jsonShorthand) {
+function getSchemaAnyway (schema, jsonShorthand, statusCode) {
   if (!jsonShorthand || schema.$ref || schema.oneOf || schema.allOf || schema.anyOf || schema.$merge || schema.$patch) return schema
-  if (!schema.type && !schema.properties) {
+  if (!schema.type && (!schema.properties && statusCode !== '204')) {
```

Comment:
Here we can consider the structure of the JSONSchemaDraft7

On my PR I was working on something like considering the keys
```js
const draft7Keys = ["$comment", "$id", "$ref", "$schema", "additionalItems", "additionalProperties", "allOf", "anyOf", "const", "contains", "contentEncoding", "contentMediaType", "default", "definitions", "dependencies", "description", "else", "enum", "examples", "exclusiveMaximum", "exclusiveMinimum", "format", "if", "items", "maxItems", "maxLength", "maxProperties", "maximum", "minItems", "minLength", "minProperties", "minimum", "multipleOf", "not", "oneOf", "pattern", "patternProperties", "properties", "propertyNames", "readOnly", "required", "then", "title", "type", "uniqueItems"] 
```

This could be an idea but I don't know if others agree with me

Reply:
@Eomm the main thing about your suggestion is that I'm missing the point when we pass from the `getSchemaSerializer` function in case of empty schemas

Reply:
@giuliowaitforitdavide Thanks!
@Eomm I read your suggestion, as I understand we call getSchemaSerializer on income request, the problem here is that we fail to start the server.

Reply:
I will check later - I think we run that function (or similar) during the startup to compile the serialize function.

Reply:
Found, here is where the magic happen:
https://github.com/fastify/fastify/blob/5b7fd1835cc4f8efb52a6946677da37d8c1561c1/lib/validation.js#L23

if we skip adding a function to the `acc`(accumulator) the response payload will not be blocked

I would add a test with a response schema with a typo on the schema, such as `propppertie`

Reply:
@Eomm Thanks, i'll check that later. 
But do we want to skip the whole schema check?